File: 2021/11/29/water_bottles.js
/**
 * https://leetcode.com/problems/water-bottles/
 * @author Andrew Jarombek
 * @since 11/29/2021
 */

const numWaterBottles = function(numBottles, numExchange) {
    let numEmpties = 0;
    let result = 0;

    while (numBottles > 0) {
        result += numBottles;

        numEmpties += numBottles;
        numBottles = 0;

        const exchange = numEmpties - (numEmpties % numExchange);
        numEmpties %= numExchange;
        numBottles = exchange / numExchange;
    }

    return result;
};

File: 2022/09/16/java/WaterBottles.java
/*
 * https://leetcode.com/problems/can-make-arithmetic-progression-from-sequence/
 * @author Andrew Jarombek
 * @since 9/16/2022
 */

public class WaterBottles {
    public static int numWaterBottles(int numBottles, int numExchange) {
        int result = 0;

        while (numBottles >= numExchange) {
            int quotient = numBottles / numExchange;
            int remainder = numBottles % numExchange;

            result += numBottles - remainder;
            numBottles = quotient + remainder;
        }

        return result + numBottles;
    }

    public static void main(String[] args) {
        assert numWaterBottles(9, 3) == 13;
        assert numWaterBottles(15, 4) == 19;
    }
}
